spec/requests/api/v0.1/source_types_spec.rb

```diff
@@ -1,5 +1,5 @@
 RSpec.describe("v0.1 - SourceTypes") do
-  let(:attributes)      { {"name" => "test_name", "product_name" => "Test Product", "vendor" => "TestVendor"} }
+  let(:attributes)      { {"name" => "test_name", "product_name" => "Test Product", "vendor" => "TestVendor", "created_at"=>"2019-02-13T16:23:24Z", "updated_at"=>"2019-02-13T16:23:24Z"} }
```

Comment:
I don't think these should be added to the attributes.  This is used for record creation and I don't think we should ever pass values for these columns.

Reply:
ok, so I'll need to tweak the asserts

Reply:
hm right, this also means that these columns should always be readOnly :-)

Reply:
@bdunne hm, I've added the readOnly and the spec is still passing. The spec for `post` should fail right? Since we should filter out the `readOnly` attrs?

Reply:
We don't fail for read-only attributes on create.  There's no concept in swagger for things allowed for create, but read-only after that.  We would have to duplicate every object definition that allows for create and didn't want the overhead right now.

Reply:
hm ok, that make sense, we might need to have different concept for non writable cols (like a separate definition for writable, then readable will inherit that and add more?)

Reply:
Don't forget about PATCH / update